src/os/ui/featureedit.js

```diff
@@ -696,6 +702,21 @@ os.ui.FeatureEditCtrl.prototype.isEllipse = function() {
 };
 
 
+/**
+ * If a line or polygon is selected.
+ * @return {boolean}
+ * @export
+ */
+os.ui.FeatureEditCtrl.prototype.isPolygonOrLine = function() {
+  var geometry = this.previewFeature.getGeometry();
+
+  return (
+    geometry instanceof ol.geom.Polygon || geometry instanceof ol.geom.MultiPolygon ||
```

Comment:
Use `geometry.getType()` (considering that the geometry could be null/undefined) and `ol.geom.GeometryType` to test the type. `instanceof` is much slower and unreliable if the geometry was created in another window context.

Reply:
I just realized that I don't actually use this function ... I guess it makes sense to have for features, but I wasn't sure if you would always know if a feature is a polygon or line (ie a geometry collection)

Reply:
If the function isn't used, you can remove it. We already have `os.geo.isGeometryPolygonal` for detecting polygons, and line detection should go in the same place (if/when needed). For collections, you can iterate over geometries in the collection and determine if it contains lines/polygons. Depending on the use case you might want at least one line/polygon, all lines/polygons, or to ignore collections entirely, so that shouldn't be done in these functions.